x-pack/plugins/maps/public/classes/styles/vector/components/symbol/custom_icon_modal.tsx

```diff
@@ -0,0 +1,411 @@
+        paddingSize="xs"
+      >
+        <EuiPanel color="subdued" paddingSize="s">
+          <EuiFlexGroup justifyContent="flexEnd" gutterSize="xs">
+            <EuiFlexItem grow={false}>
+              <EuiButtonEmpty size="xs" onClick={this._resetAdvancedOptions}>
+                {strings.getResetButtonLabel()}
+              </EuiButtonEmpty>
+            </EuiFlexItem>
+          </EuiFlexGroup>
+          <EuiFormRow
+            className="mapsCustomIconForm__cutoff"
+            label={strings.getCutoffRangeLabel()}
+            display="rowCompressed"
+          >
+            <ValidatedRange
+              min={0}
+              max={1}
+              value={cutoff}
+              step={0.01}
+              showInput
+              showLabels
+              compressed
+              className="mapsCutoffRange"
+              onChange={this._handleCutoffChange}
+            />
+          </EuiFormRow>
+          <EuiFormRow
+            className="mapsCustomIconForm__radius"
+            label={strings.getRadiusRangeLabel()}
+            display="rowCompressed"
+          >
+            <ValidatedRange
+              min={0}
+              max={1}
+              value={radius}
+              step={0.01}
+              showInput
+              showLabels
+              compressed
+              className="mapsRadiusRange"
+              onChange={this._handleRadiusChange}
+            />
+          </EuiFormRow>
+        </EuiPanel>
+      </EuiAccordion>
+    );
+  }
+
+  private _renderIconForm() {
+    const { label, svg } = this.state;
+    return svg !== '' ? (
+      <>
+        <EuiFormRow
+          label={strings.getNameInputLabel()}
+          helpText={strings.getCharactersRemainingDescription(MAX_LABEL_LENGTH - label.length)}
+          display="rowCompressed"
+        >
+          <EuiFieldText
+            value={label}
+            className="mapsCustomIconForm__label"
+            onChange={(e) => this._handleChange('label', e.target.value)}
+            required
+            data-test-subj="mapsCustomIconForm-label"
+          />
+        </EuiFormRow>
+        <EuiSpacer />
+        {this._renderAdvancedOptions()}
+      </>
+    ) : null;
+  }
+
+  private _renderIconPreview() {
+    const { svg, isFileInvalid, cutoff, radius } = this.state;
+    return svg !== '' ? (
+      <EuiFlexItem className="mapsIconPreview__wrapper mapsCustomIconForm__preview" grow={false}>
+        <IconPreview svg={svg} isSvgInvalid={isFileInvalid} cutoff={cutoff} radius={radius} />
+      </EuiFlexItem>
+    ) : null;
+  }
+
+  public render() {
+    const { onSave, onCancel, onDelete, title, ...rest } = this.props;
+    const { symbolId, label, svg, cutoff, radius, isFileInvalid } = this.state;
+    const isComplete = label.length !== 0 && svg.length !== 0 && !isFileInvalid;
+    const fileError = svg && isFileInvalid ? strings.getInvalidFileLabel() : '';
+    return (
+      <EuiModal
+        {...rest}
```

Comment:
Why are you spreading rest here. This is props in the dom like `svg`. I don't think this is needed.
<img width="500" alt="Screen Shot 2022-03-24 at 4 01 32 PM" src="https://user-images.githubusercontent.com/373691/160017686-5e4dd0a8-0acc-4d27-aa8f-efbc6abb32b0.png">